### ISSUE TITLE
Make options ordering more clear

### ISSUE THREAD

**Issue by jtmarmon** [Yuppiechef/datomic-schema#12]
I had to do a big of debugging to understand why

```clj
(fields
    [thing :many :string])
```

Was invalid. Didn't realize until I took a look at the source. I would add a PR for the README explaining that opts are last, but perhaps it makes more send to have options in hash w/ boolean values or a nested vector or something. What do you think?

**Comment by CmdrDats** [comment]
The alternative would be to include the type as just another 'option' so that you really can have it wherever. Since there aren't bound to be too many name clashes with other options, this should be fine.

This might lead to fragmented ordering (some people ordering the type last, and others first, and yet others arbitrarily), but I don't reckon that's really a problem.

I had thought of putting the options into a set, tbh, but I didn't particularly like the levels of nesting that came out of `[thing :enum #{[:a :b :c] :many}]`, or even the simpler case of `[thing :string #{:many}]`, since seeing that over hundreds of fields might just make your eyes bleed.

**Comment by jtmarmon** [comment]
yeah i agree it's prettier on the eyes for sure. just needs a better error then and some documentation :) can't get to either issue right now but I can likely sometime this week!
